packages/syft/src/syft/core/node/common/node_service/oblv/oblv_service.py

```diff
@@ -0,0 +1,769 @@
+"""
+This file defines all the functions/classes to perform oblv actions, for a given domain node, in a RESTful manner.
+"""
+
+# stdlib
+from base64 import encodebytes
+import os
+from os import path
+import subprocess  # nosec
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import Type
+from typing import Union
+
+# third party
+from nacl.signing import VerifyKey
+from oblv import OblvClient
+import requests
+
+# relative
+from ......logger import debug
+from .....common.message import ImmediateSyftMessageWithReply
+from .....common.message import ImmediateSyftMessageWithoutReply
+from .....common.serde.serialize import _serialize as serialize
+from .....common.uid import UID
+from ....domain_interface import DomainInterface
+from ...exceptions import AuthorizationError
+from ...exceptions import OblvEnclaveError
+from ...exceptions import OblvEnclaveUnAuthorizedError
+from ...exceptions import OblvProxyConnectPCRError
+from ..auth import service_auth
+from ..node_service import ImmediateNodeServiceWithReply
+from ..node_service import ImmediateNodeServiceWithoutReply
+from ..success_resp_message import SuccessResponseMessage
+from .oblv_messages import CheckEnclaveConnectionMessage
+from .oblv_messages import CreateKeyPairMessage
+from .oblv_messages import DeductBudgetMessage
+from .oblv_messages import GetPublicKeyMessage
+from .oblv_messages import GetPublicKeyResponse
+from .oblv_messages import PublishApprovalMessage
+from .oblv_messages import PublishDatasetMessage
+from .oblv_messages import PublishDatasetResponse
+
+USER_INPUT_MESSAGES = Union[
+    GetPublicKeyMessage,
+    PublishDatasetMessage,
+    CheckEnclaveConnectionMessage,
+    CreateKeyPairMessage,
+]
+
+USER_OUTPUT_MESSAGES = Union[SuccessResponseMessage, GetPublicKeyResponse]
+
+
+def create_keys_from_db(node):
+    file_path = os.getenv("OBLV_KEY_PATH", "/app/content")
+    file_name = os.getenv("OBLV_KEY_NAME", "oblv_key")
+    keys = node.oblv_keys.get()
+
+    # Creating directory if not exist
+    os.makedirs(
+        os.path.dirname(file_path + "/" + file_name + "_private.der"), exist_ok=True
```

Comment:
MAYBE: Replace all the uses of `os.path` with Pathlib

Reply:
Yep, will create a new ticket for the same.